File: server/messagebus/commands.go
package messagebus

type Command struct {
	Party
}

func (c Command) Type() Type {
	return CommandMessage
}

type JoinParty struct {
	Command
	Player string
}

type StartGame struct {
	Command
}

type LeaderSelectsMember struct {
	Command
	Leader         string
	MemberToSelect string
}

type LeaderDeselectsMember struct {
	Command
	Leader           string
	MemberToDeselect string
}

type LeaderConfirmsTeamSelection struct {
	Command
	Leader string
}

type ApproveTeam struct {
	Command
	Player string
}

type RejectTeam struct {
	Command
	Player string
}

type SucceedMission struct {
	Command
	Player string
}

type FailMission struct {
	Command
	Player string
}

File: client/src/messages/commands.ts
import type { Message } from "./message-bus";

export class JoinParty implements Message {
  constructor(readonly name: string){}
}

export class StartGame implements Message {}

export class LeaderSelectsMember implements Message {
  constructor(readonly member: string){}
}

export class LeaderDeselectsMember implements Message {
  constructor(readonly member: string){}
}

export class LeaderConfirmsTeam implements Message {}

export class ApproveTeam implements Message {}

export class RejectTeam implements Message {}

export class SucceedMission implements Message {}

export class FailMission implements Message {}

export class ViewIdentity implements Message {}

export class ViewMissionDetails implements Message {
  constructor(readonly mission:number){}
}

export class CloseDialog implements Message {}
